src/js/snapping.service.js

```diff
@@ -0,0 +1,84 @@
+import * as MathService from "./math.service";
+
+export function isLeftSnapping(firstPolygon, secondPolygon) {
+    const firstLine = MathService.initializeLine(secondPolygon.vertices[1], secondPolygon.vertices[2]);
```

Comment:
how about implement at Polygon methods like `getTopSide`, `getBottomSide` and etc